deploy/kubernetes/base/node.yaml

```diff
@@ -16,7 +16,6 @@ spec:
     spec:
       nodeSelector:
         kubernetes.io/os: linux
-        kubernetes.io/arch: amd64
```

Comment:
Updating this file directly will break users who use the older driver, doesn't it?  

I have a proposal of starting branching install yaml files. Will update the link once I write it down.

Reply:
Why will it break?

IIUC your question. Applying new manifest on old driver is not expected. Each driver release comes with it's dedicated manifest and helm chart. 

User should get the manifest or chart using the version tag. eg. here is the manifest for v0.2 driver: https://raw.githubusercontent.com/kubernetes-sigs/aws-efs-csi-driver/release-0.2.0/deploy/kubernetes/manifest.yaml
And there is a v0.3.0 tag which contains to v0.3.0 driver manifest.

And the versioned chart is located at: https://github.com/kubernetes-sigs/aws-efs-csi-driver/releases/download/{version}/helm-chart.tgz

That being said, this change will only be part of v0.4.0 release, which user only suppose to apply when upgrading / creating v0.4.0 driver

Reply:
The current recommended way to install stable driver version according to the README is 

> kubectl apply -k "github.com/kubernetes-sigs/aws-efs-csi-driver/deploy/kubernetes/overlays/stable/?ref=master"

And that will pick up this code with 0.3.0 driver image. 

And that's is also why we had bug like https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/191. 

Please take a look at a proposal to solve this problem by branching the yaml install file. 

https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/207